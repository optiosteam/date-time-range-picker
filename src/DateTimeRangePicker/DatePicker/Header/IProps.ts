import moment from 'moment'
import {TCalendarMode} from '../TCalendarMode'

interface IProps {
    date: moment.Moment,
    months: number,
    onNextMonth: () => void,
    onPrevMonth: () => void
    onNextYear: () => void
    onPrevYear: () => void
    calendarMode: TCalendarMode
    setCalendarMode: (mode: TCalendarMode) => void
}

export default IProps
