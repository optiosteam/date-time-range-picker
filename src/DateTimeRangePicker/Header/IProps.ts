import moment from 'moment'

interface IProps {
    date: moment.Moment,
    months: number
}

export default IProps
