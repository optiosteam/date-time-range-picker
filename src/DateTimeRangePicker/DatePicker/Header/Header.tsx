import React from 'react'

import IProps from './IProps'

const Header: React.FunctionComponent<IProps> = ({date, months, onPrevMonth, onNextMonth}) => {
    const monthsArray = []
    for (let monthIndex = 0; monthIndex < months; monthIndex++) {
        monthsArray.push(monthIndex)
    }

    return <div className={'date-time-range-picker-header'}>
        <button className={'date-time-range-picker-header-prev'} onClick={onPrevMonth}>&laquo;</button>
        {monthsArray.map(
            (monthIndex: number) => {
                return <div className={'date-time-range-picker-header-month'} key={`monthHeader${monthIndex}`}>
                    {date.clone().add(monthIndex, 'months').format('MMMM YYYY')}
                </div>
            }
        )}

        <button className={'date-time-range-picker-header-next'} onClick={onNextMonth}>&raquo;</button>
    </div>
}

export default Header
