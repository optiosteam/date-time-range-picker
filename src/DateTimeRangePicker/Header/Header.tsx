import React, {ReactElement} from 'react'

import IProps from './IProps'

const Header: React.FunctionComponent<IProps> = ({date, months}) => {
    const monthsArray = []
    for (let monthIndex = 0; monthIndex < months; monthIndex++) {
        monthsArray.push(monthIndex)
    }

    return <div className={'date-time-range-picker-header'}>
        <button className={'date-time-range-picker-header-prev'}>&laquo;</button>
        {monthsArray.map(
            (monthIndex: number) => {
                return <div className={'date-time-range-picker-header-month'} key={`monthHeader${monthIndex}`}>
                    {date.clone().add(monthIndex, 'months').format('MMMM')}
                </div>
            }
        )}

        <button className={'date-time-range-picker-header-next'}>&raquo;</button>
    </div>
}

export default Header
