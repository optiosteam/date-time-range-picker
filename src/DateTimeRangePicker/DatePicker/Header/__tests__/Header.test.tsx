 import React from 'react'
 import { shallow } from 'enzyme'
 import toJson from 'enzyme-to-json'

 import Header from '../Header'
 import moment from 'moment'

 describe('Date/time range picker header', () => {
    it('should render correctly', () => {
        const component = shallow(
            <Header date={moment('2019-10-15 18:23:15', 'YYYY-MM-DD HH:mm:ss')}
                    months={1}
                    onPrevMonth={() => { return }}
                    onNextMonth={() => { return }}
                    onNextYear={() => { return }}
                    onPrevYear={() => { return }}
                    calendarMode={'normal'}
                    setCalendarMode={() => { return }}
            />
        )

        expect(toJson(component)).toMatchSnapshot()
    })

    it('should call function on month navigation', () => {
        const onPrevMonthHandler = jest.fn()
        const onNextMonthHandler = jest.fn()

        const component = shallow(
            <Header date={moment('2019-10-15 18:23:15', 'YYYY-MM-DD HH:mm:ss')}
                    months={1}
                    onPrevMonth={onPrevMonthHandler}
                    onNextMonth={onNextMonthHandler}
                    onNextYear={() => { return }}
                    onPrevYear={() => { return }}
                    calendarMode={'normal'}
                    setCalendarMode={() => { return }}
            />
        )
        component.find('.date-time-range-picker-header-prev').simulate('click')
        component.find('.date-time-range-picker-header-next').simulate('click')
        component.find('.date-time-range-picker-header-next').simulate('click')
        expect(onPrevMonthHandler).toBeCalledTimes(1)
        expect(onNextMonthHandler).toBeCalledTimes(2)
    })
})
