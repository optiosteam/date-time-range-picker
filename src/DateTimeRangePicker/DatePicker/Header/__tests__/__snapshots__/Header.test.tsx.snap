// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date/time range picker header should render correctly 1`] = `
<div
  className="date-time-range-picker-header"
>
  <button
    className="date-time-range-picker-header-prev"
    onClick={[Function]}
    type="button"
  >
    «
  </button>
  <div
    className="date-time-range-picker-header-month"
    key="monthHeader0"
  >
    October 2019
  </div>
  <button
    className="date-time-range-picker-header-next"
    onClick={[Function]}
    type="button"
  >
    »
  </button>
</div>
`;
